processing/data_collection/gazette/pipelines.py

```diff
@@ -141,3 +143,37 @@ def _is_file_type(self, filepath, file_types):
         Generic method to check if a identified file type matches a given list of types
         """
         return self.get_file_type(filepath) in file_types
+
+
+class RequestWithItem(Request):
+    """
+    Specialized Request object to allow carry the item which generate the request.
+    Thus, we can use the gazette date in the path where the file will be stored.
+    """
+
+    def __init__(self, url, item):
+        super().__init__(url)
+        self.item = item
+
+
+class QueridoDiarioFilesPipeline(FilesPipeline):
+    """
+    When the downloaded file are stored in a remote storage system (e.g.
+    Digital Ocean spaces), we need to specialize FilesPipeline class in order
+    to allow us define a different directory where the files will be store. In
+    the current implementation we organize gazette files by date. All the
+    gazettes from the same date will be store in the same directory.
+    """
+
+    def file_path(self, request, response=None, info=None):
+        filepath = super().file_path(request, response, info)
+        datestr = request.item["date"].strftime("%d-%m-%Y")
+        # The default path from the scrapy class begins with "full/". In this
+        # class we replace that with the gazette date.
+        return str(pathlib.Path(datestr, filepath[5:]))
```

Comment:
Em Python, [a legibilidade importa](https://www.python.org/dev/peps/pep-0020/). usar `filepath[:5]` para retirar o `full/` do início de um caminho de arquivo prejudica a legibilidade pois é uma escolha de uma estratégia de baixo nível de abstração — quando o Python oferece abstrações de mais alto nível, por exemplo:

```python
In [1]: from pathlib import Path                                                                                                                                                                                     

In [2]: path = Path('full/abcd.ef')                                                                                                                                                                                  

In [3]: path.name                                                                                                                                                                                                    
Out[3]: 'abcd.ef'
```

Acho que eliminando o _slice_ de uma _string_ e usando uma abstração como o `pathlib.Path` conseguimos deixar esse código mais legível e compreensível… o que acha?